components/eamxx/src/share/util/eamxx_time_interpolation.cpp

```diff
@@ -268,6 +266,19 @@ void TimeInterpolation::set_file_data_triplets(const vos_type& list_of_files) {
     } else {
       EKAT_ERROR_MSG("Error!! TimeInterpolation::set_file_triplets - unsupported units of time = (" << time_units << ") in source data file " << filename << ", supported units are: seconds, minutes, hours and days");
     }
+    auto date_start = time_units.find_first_of("0123456789");
+    auto time_units_date = time_units.substr(date_start);
+    // Parse the time_units_date into a vector of date and time
+    // Reference TimeStamp
+    auto tokens = ekat::split(time_units_date," ");
+    auto date_tkn = ekat::split(tokens[0],"-");
+    auto time_tkn = ekat::split(tokens[1],":");
+    std::vector<int> ts_date, ts_time;
+    for (int jj=0; jj<3; jj++) {
+      ts_date.push_back(stoi(date_tkn[jj]));
+      ts_time.push_back(stoi(time_tkn[jj]));
+    } 
+    const TimeStamp ts_file_start(ts_date,ts_time);
```

Comment:
Is this really a file start? The units will likely be "nXYZ from <case t0>", so even if you have multiple files, they should all have the same units, no? Perhaps you should add the first value of the time var to `ts_ref`?

Reply:
Are you saying we could have a list of files with different units? If we allow this, for safety, I think we should check that the files are "not overlapping", which is kind of tedious/annoying.

Reply:
Well, I see both a different case_t0 as well as a different scale (minutes vs days).. I'm ok leaving to the user the burden of verifying nonoverlap-ness, but mixing units may be confusing...